src/Services/Ordering/Ordering.Domain/SeedWork/Enumeration.cs

```diff
@@ -11,27 +11,22 @@ public abstract class Enumeration : IComparable
 
         public int Id { get; private set; }
 
-        protected Enumeration(int id, string name)
-        {
-            Id = id;
-            Name = name;
-        }
+        protected Enumeration(int id, string name) => (Id, Name) = (id, name);
 
         public override string ToString() => Name;
-
-        public static IEnumerable<T> GetAll<T>() where T : Enumeration
-        {
-            var fields = typeof(T).GetFields(BindingFlags.Public | BindingFlags.Static | BindingFlags.DeclaredOnly);
-
-            return fields.Select(f => f.GetValue(null)).Cast<T>();
-        }
-
+        
+        public static IEnumerable<T> GetAll<T>() where T : Enumeration =>
+            typeof(T).GetFields(BindingFlags.Public |
+                                BindingFlags.Static |
+                                BindingFlags.DeclaredOnly)
+                     .Select(f => f.GetValue(null)).Cast<T>();
+        
         public override bool Equals(object obj)
         {
-            var otherValue = obj as Enumeration;
-
-            if (otherValue == null)
+            if (!(obj is Enumeration otherValue))
```

Comment:
```suggestion
            if (obj is not Enumeration otherValue)
```

Reply:
This will require bumping the project from 9.0 - is that OK for the scope of this issue?

Reply:
Hi @mfcallahan, we are in process of updating this project to .NET 5 and C# 9.0. It should be available soon. 
So to my opinion, it would be better to use C# 9.0 feature itself, because it would save some rework in both from docs and reference app point of view. We will merge this PR once the `dev` branch is updated and performed the integration testing then. Thoughts ?

Reply:
Sounds good, I will commit @IEvangelist's suggestion here and use the newer C# feature knowing the project is in the process of being updated and this pull request will be merged at that time.

Reply:
We should really try to keep this at least cross-referenced... https://github.com/dotnet/docs/pull/21677

Reply:
Yes, that's right. I have updated the label also to `.NET 5.0` for better clarity.

Reply:
Hi @mfcallahan, `dev` branch is updated to `.NET 5` now. Could you please commit the suggestion and update this PR ? Will test and merge in the changes.

Thank you.